src/main/java/org/opennms/nephron/catheter/Simulation.java

```diff
@@ -271,25 +188,30 @@ public long getBytesSent() {
         return this.bytesSent;
     }
 
+    public Random getRandom() {
+        return this.random;
+    }
+
+    public BiConsumer<Exporter, FlowReport> getHandler() {
+        return this.handler;
+    }
+
     public static class Builder {
-        public String flowTopic;
         public long seed = new Random().nextLong();
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [DMI: Random object created and used only once (DMI_RANDOM_USED_ONLY_ONCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dmi-random-object-created-and-used-only-once-dmi-random-used-only-once).
This code creates a java.util.Random object, uses it to generate one random number, and then discards the Random object. This produces mediocre quality random numbers and is inefficient. If possible, rewrite the code so that the Random object is created once and saved, and each time a new random number is required invoke a method on the existing Random object to obtain it.
If it is important that the generated Random numbers not be guessable, you *must* not create a new Random for each random number; the values are too easily guessable. You should strongly consider using a java.security.SecureRandom instead (and avoid allocating a new SecureRandom for each random number needed).